Src/DependencyCollector/Shared/Implementation/SqlClientDiagnosticSourceListener.cs

```diff
@@ -0,0 +1,488 @@
+                        (IsolationLevel)TransactionAfter.IsolationLevel.Fetch(evnt.Value),
+                        (SqlConnection)TransactionAfter.Connection.Fetch(evnt.Value),
+                        (long)TransactionAfter.Timestamp.Fetch(evnt.Value));
+
+                    break;
+                }
+
+                case SqlErrorCommitTransaction:
+                case SqlErrorRollbackTransaction:
+                {
+                    this.OnErrorTransactionEvent(
+                        (Guid)TransactionError.OperationId.Fetch(evnt.Value),
+                        (string)TransactionError.Operation.Fetch(evnt.Value),
+                        (IsolationLevel)TransactionError.IsolationLevel.Fetch(evnt.Value),
+                        (SqlConnection)TransactionError.Connection.Fetch(evnt.Value),
+                        (Exception)TransactionError.Exception.Fetch(evnt.Value),
+                        (long)TransactionError.Timestamp.Fetch(evnt.Value));
+
+                    break;
+                }
+            }
+        }
+        
+        private void OnAfterCommandEvent(
+            Guid operationId,
+            string operation,
+            Guid connectionId,
+            SqlCommand command,
+            IDictionary statistics,
+            long endTimestamp)
+        {
+            var dependencyName = string.Empty;
+            var target = string.Empty;
+
+            if (command.Connection != null)
+            {
+                target = string.Join(" | ", command.Connection.DataSource, command.Connection.Database);
+
+                var commandName = command.CommandType == CommandType.StoredProcedure
+                    ? command.CommandText
+                    : string.Empty;
+
+                dependencyName = string.IsNullOrEmpty(commandName)
+                    ? string.Join(" | ", command.Connection.DataSource, command.Connection.Database)
+                    : string.Join(" | ", command.Connection.DataSource, command.Connection.Database, commandName);
+            }
+
+            var duration = this.DeriveDuration(operationId, endTimestamp);
+
+            var telemetry = new DependencyTelemetry()
+            {
+                Id = operationId.ToString(),
+                Name = dependencyName,
+                Type = RemoteDependencyConstants.SQL,
+                Target = target,
+                Data = command.CommandText,
+                Duration = duration,
+                Timestamp = DateTimeOffset.UtcNow - duration,
+                Success = true
+            };
+
+            this.client.Track(telemetry);
+        }
+
+        private void OnCommandErrorEvent(
+            Guid operationId,
+            string operation,
+            Guid connectionId,
+            Exception exception,
+            long endTimestamp)
+        {
+            var duration = this.DeriveDuration(operationId, endTimestamp);
+
+            // TODO: How should this be correlated to the parent dependency event?
```

Comment:
ApplicationInsights will set telemetry.Operation.Context.Id and ParentId when telemetry is initialized.

However, since there was no Activity started for parent dependency, exception will be a sibling to the dependency.
You have some operationId in the event. Is it id of the query/transaction?
If so, you can just set telemetry.Operation.Context.Parent.Id to it and it will be correlated to the parent dependency.